h2o-core/src/main/java/water/parser/SyntheticColumnGenerator.java

```diff
@@ -0,0 +1,63 @@
+package water.parser;
+
+import water.Job;
+import water.Key;
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+
+public class SyntheticColumnGenerator extends MRTask<SyntheticColumnGenerator> {
+  
+  public static Frame addSyntheticColumns(Job<Frame> job, Vec[] parsedVecs, ParseSetup setup, Key[] fkeys, int[] fileChunkOffsets) {
```

Comment:
I would like to have this decoupled from ParseDataset 

this call is confusing to the reader:

      - fr = new Frame(job._result, setup._column_names, AppendableVec.closeAll(avs));
      + fr = SyntheticColumnGenerator.addSyntheticColumns(job, AppendableVec.closeAll(avs), setup, fkeys, mfpt._fileChunkOffsets);

because it in the common case only creates the frame without any synthetics columns

you can have an interface, called eg. ParseFinalizer that would have have 2 implementations based on `setup._synthetic_column_names` with a factory-method somewhere